File: projects/caseStudyMicroservices/hotel-service/src/main/java/com/sourabh/hotelservice/repository/RoomRepository.java
package com.sourabh.hotelservice.repository;


import com.sourabh.hotelservice.model.Room;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;


@Repository
public interface RoomRepository extends MongoRepository<Room,Integer> {

    Room findByType(String type);
    @Query(value="{available:true}")//,fields="{_id:0,}")
    List<Room> getAllAvailable();
}

File: projects/spring-boot-rest-kt/hotel-reservation/src/main/kotlin/com/sourabh/hotelreservation/repository/RoomRepository.kt
package com.sourabh.hotelreservation.repository

import com.sourabh.hotelreservation.model.Room
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface RoomRepository:MongoRepository<Room,Int> {
    fun findByType(type:String):Room?
    @Query(value="{available:true}")
    fun getAllAvailable():MutableList<Room>
}